File: 0001-two-sum/0001-two-sum.rs
use std::collections::HashMap;

impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let mut hash_map: HashMap<i32, i32> = HashMap::new();
        for i in 0..nums.len() {
            let left_over = target - nums[i];
            let data = hash_map.get_key_value(&left_over);
            if data.is_some() {
                return vec![i as i32, *(data.unwrap().1)]
            }
            hash_map.insert(nums[i], i as i32);
        }
        vec![]
    }
}

File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        std::unordered_map<int, int> map;

        for (int i = 0; i < nums.size(); i++) {
            int val = target - nums[i];
            if (map.find(val) != map.end()) {
                return {map[val], i};
            }
            map[nums[i]] = i;
        }

        return {0,0};
    }
};